### DISCUSSION TITLE
Collaboration Tool

### DISCUSSION THREAD

**Discussion by dsteuart** [opentoonz/opentoonz#6511] [Ideas]
### Is your feature request related to a problem? Please describe

A tool where people can colloborate at the same time on a project in OpenToonz

### Describe the solution you'd like

A new tool or some type of cloud space

**Comment by RodneyBaker** [comment]
@dsteuart Thanks for the request.
I agree that collaboration tools are important and will be increasingly so in the future.

The primary collaboration tool Opentoonz has at present is a set of tools for connecting to SVN which is a centralized versioning system similar to and (mostly) compatible with git/github.  REF:  https://opentoonz.readthedocs.io/en/latest/using_the_version_control.html

Note that I assume the SVN feature does not currently meet your requirement.

As such...
For discussions sake, your request should probably go beyond the generic nature of collaboration tools and specify what tools in particular you personally would prioritize if/when such tools are to be implemented.  While it may be awhile before Opentoonz gains those tools natively there are also likely many ways to use collaboration tools with Opentoonz even now.  Knowing more about your specific needs will help others get that properly addressed.**

You do mention 'cloud space'.
I am curious, if such a space were available would you be willing to pay something for that service?
I ask because things that can be funded might gain support and be developed at a faster pace.

In general, and without regard to any potential cost, what does success look like to you?  
How will developers know when they have met your requirement for the proposed collaboration tool(s)?


**My personal research related to collaboration thus far has mostly focused on Peer to Peer (P2P) solutions which unfortunately have not found a way into the mainstream.  My take is that the reason it hasn't been made to work is that no one has found a way to profit off of P2P solutions in a meaningful way and legitimately it does take money to keep services going even with P2P approaches which most would consider free services.  The most promising of my leads actually shut down so... that general approach ([LINK](https://beakerbrowser.com/)) obviously and unfortunately isn't going to be a path to success.  ;)   (Note for developers:  The underlying protocol behind Beaker Browser was (at least for most of its life hypercore which is still somewhat active.  REF:  https://github.com/hypercore-protocol/hypercore)

**Comment by RodneyBaker** [comment]
Recommend some deep research be done concerning the state of the art in collaboration tools (both free and paid).
Two primary areas of interest:

Internal
What collaborative areas can be maximized inside of Opentoonz to provide a framework for integrating more fully with collaborative tools

External 
What collaborative programs and services can be leveraged to fully exploit user collaboration

The way forward would appear to be use of external collaborative tools primarily for the following reasons:

- Number of developers available and able to commit to collaboration system(s).
- Extent of management and maintenance required to keep collaborative tools maintained and relevant
- Those with the greatest need should reasonably be expected to support these efforts with feedback, funding and demonstrations of level of success.  For instance, studios would likely require dedicated personnel to tailor collaborative tools to the needs of studio users and production requirements.  Individual users would need to know how best to connect and collaborate with others.
- Security and Confidentiality must be considered.  
- Etc.

It does not seem reasonable that Opentoonz will be able to gain and support unique collaborative tools not common to the needs of other programs.  Combine this with the high likelihood of other collaborators requiring and even preferring other applications in their pipeline and we can easily see where Opentoonz would be one node in a network of a larger collaborative ecosystem.

It is important to determine what collaborative tools and requirements are unique to Opentoonz alone and in what ways those might be made less unique without losing the benefits and strengths of those for Opentoonz users.

Much of this can be researched (and developed) at the user level as the user of Opentoonz is the best determiner of what they need to effectively collaborate with other Opentoonz users (and the external world).

Something that would be helpful regarding collaborative tools is a determination of what current and requested features already have collaborative capabilities and how can they be further improved to meet collaboration requirements in the future.

As for this specific request for a collaboration tool it is entirely too vague to act upon in any meaningful way.
As such additional information is required to narrow the scope of the request and to focus on exact collaborative tools required moving forward.

Finally, given the perfect collaborative tool(s) for Opentoonz may not be available any time soon, what alternatives are there currently to allow Opentoonz users to collaborate NOW versus when those tools finally become available to the satisfaction of all users. 

> A tool where people can colloborate at the same time on a project in OpenToonz

> A new tool or some type of cloud space

@dsteuart It will help tremendously if you can describe this proposed tool in any further detail.
Can you provide links or examples of what success should look like?

**Comment by RodneyBaker** [comment]
Curious what others are using currently for collaboration.
@dsteuart How are you collaborating currently and what tools do you find work best for you?
